### PR TITLE
feat(hog): hog template string editor

### PR DESCRIPTION
**Pull request by mariusandra**
~Depends on https://github.com/PostHog/posthog/pull/23232~

## Problem

Hog template string editing is meh

## Changes

- Adds Hog expression and Hog template autocomplete
- Creates `<CodeEditorInline />` that is an inline monaco with all autocomplete and code validation features
- Changes the DeliveryHog's string template editors into those new tags
- Manual focus control to make the `<tab>` key work with inline code editors
- Fix a few random mypy errors that flared up in tests
- Removed HogQL printer tests from depending on `translate_hogql` - it's a function that is only used in old insights
- Metadata for template strings

## How did you test this code?

Added tests, see screencast

## Out of scope

- Full Hog autocomplete
- Updating HogQL filters with new inline editor (easy to add, but maybe slows things down and we want to revert --> another PR)

### CODE REVIEW

File: frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx [hunk 1]
```diff
@@ -240,7 +241,14 @@ function DictionaryField({ onChange, value }: { onChange?: (value: any) => void;
 export function HogFunctionInputRenderer({ value, onChange, schema, disabled }: HogFunctionInputProps): JSX.Element {
     switch (schema.type) {
         case 'string':
-            return <LemonInput value={value} onChange={onChange} className="ph-no-capture" disabled={disabled} />
+            return (
+                <CodeEditorInline
```
Comment (benjackwhite) [inline frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx:245]: The only thing that I think doesn't work here is that the autocomplete is based on the Clickhouse style event structure which is different to the structure used by HogFunctions.

Needs to be structured like the json editor https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx#L47

**Review by benjackwhite — APPROVED** [review]
Generally this all looks great to me. Happy for the comment about the wrong data structure to be addressed in follow up if you think that makes sense

File: frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx [hunk 2]
```diff
@@ -201,15 +202,15 @@ function DictionaryField({ onChange, value }: { onChange?: (value: any) => void;
                         placeholder="Key"
                     />
 
-                    <LemonInput
-                        className="flex-1"
+                    <CodeEditorInline
```
Comment (benjackwhite) [inline frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx:205]: One other small thing.

Would be cool if we could make it even more subtle and get rid of the right click behaviour. And maybe have a less "code-y" font (but thats not so important)
Screenshot 2024-06-26 at 09 21 27

File: frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx (hunk 1 above)
Comment (mariusandra) [reply frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx:245]: Yeah, this felt like something I could push to the next PR as this got big/long enough. Will definitely fix.

File: frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx (hunk 2 above)
Comment (mariusandra) [reply frontend/src/scenes/pipeline/hogfunctions/HogFunctionInputs.tsx:205]: Will address in a follow up. Can't waste the green state of this PR... 😅

**Pull request merged**